.php-cs-fixer.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+use PhpCsFixer\Config;
+use PhpCsFixer\Finder;
+
+/**
+ * Read excluded paths from .gitignore
+ *
+ * @param string $dir
+ *
+ * @return string[]
+ */
+function getGitignorePaths(string $dir): array
+{
+    $gitignoreFile = $dir . '/.gitignore';
+    $paths         = [];
+
+    if (!file_exists($gitignoreFile)) {
+        return $paths;
+    }
+
+    $lines = file($gitignoreFile, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
+    if (!empty($lines)) {
+        foreach ($lines as $line) {
+            // Ignore comments and empty lines
+            if (strpos($line, '#') === 0 || trim($line) === '') {
+                continue;
+            }
+            // Add relative paths
+            $paths[] = trim($line);
+        }
+    }
+
+    return $paths;
+}
```

Comment:
According to the [documentation](https://cs.symfony.com/doc/config.html):

> Default finder ignores __DIR__ . "/vendor" dir, “hidden” paths (ones starting with a dot) and VCS paths (e.g. .git), and filter only for *.php files.

What kind of files do you want to exclude with this function ? Most of the time, the defaut exclusions will do the job, and manually adding a specific dir is probably simplier than maintaining this function.

Reply:
I initially intended to create a generic configuration that could be identical across each plugin, allowing for simple copy-and-paste. However, it might not be optimal, so I will remove it.